docs/modules/ROOT/pages/explanations/decisions/syn-argocd-sharing.adoc

```diff
@@ -0,0 +1,63 @@
+* Teams can manage their ArgoCD applications on a cluster without interfering with other teams' ArgoCD applications
+
+=== Non-Goals
+
+* Allow deploying arbitrary applications through the Project Syn ArgoCD instance
+* Create a well-defined structure for per-team configurations in the Project Syn tenant repository
+
+== Proposals
+
+=== Add team label to ArgoCD apps
+
+We can keep the current setup (one ArgoCD instance with one https://argo-cd.readthedocs.io/en/stable/operator-manual/cluster-bootstrapping/#app-of-apps-pattern[root application ("app of apps")]) and inject additional information in the form of labels on the ArgoCD applications.
+
+This approach can easily be tested by injecting suitable application labels manually in Commodore components.
+Additionally, engineering streamlined support for this approach is straightforward and can be done in the https://github.com/projectsyn/component-argocd/blob/1227b46c2db6e6959ce75aa7ab44db975f33bb3e/lib/argocd.libjsonnet[argocd component library] and the configuration hierarchy.
+A potential implementation would be to provide a mapping of component instance names to team names in the configuration hierarchy.
+This would enable the component library to lookup the value for the team label with minimal requirements.
+
+However, this approach doesn't address the issue that it's currently not possible to pause ArgoCD auto sync without pausing auto sync for the global root app.
+
+=== Create separate ArgoCD project and root application per team
+
+The next option is to adjust the Project Syn-managed ArgoCD to support an ArgoCD Project and root app per team.
+This approach will need more extensive engineering, since a lot of Project Syn tooling (at least https://github.com/projectsyn/steward/blob/c25ea41615340ab8b6a78888d03fe710f77d86de/pkg/argocd/argocd.go#L88-L94[Steward], https://github.com/projectsyn/component-argocd/blob/1227b46c2db6e6959ce75aa7ab44db975f33bb3e/component/app.jsonnet[component argocd] and the https://github.com/projectsyn/commodore-component-template/blob/cfd150bd41deaa0ad10d522ec5fd0abafed65800/%7B%7B%20cookiecutter.slug%20%7D%7D/class/%7B%7B%20cookiecutter.slug%20%7D%7D.yml#L4-L7[component template]) currently assumes that there's exactly one ArgoCD project and root application for all applications managed through Project Syn.
+
+With this approach, the team responsible for the cluster itself would continue using the current ArgoCD project (`syn`) and root application (`root`).
+This team can bootstrap ArgoCD projects and root applications for any other teams who deploy applications to the cluster through Project Syn.
+
+To support this approach, the argocd component library needs to be made "team-aware" (similar to the first alternative).
+Instead of using the component instance to team mapping to add labels to each ArgoCD application, the component library can ensure the correct ArgoCD project is configured for each application.
```

Comment:
I'm not entirely sure how this would fix the pausing auto sync issue. Wouldn't we still need to

pause root app -> pause team app -> pause component app

If I understand the approach correctly the root app will still deploy the "team" app so to disable auto sync for the team app we'd still need to disable auto sync for the root. Or should the root app not manage sync options of the team app?

Reply:
The idea is that each team app would be deployed as an additional root app in a separate project (i.e. the team app for `team1` would point to a separate directory in the cluster catalog where the team app itself is also stored) -- maybe that's not quite clear from the current text. Also, I'm not 100% sure how the bootstrap procedure for this will look like.

Reply:
Aahh. So we'd bootstrap a completely different root app that just happens to live in the same cluster catalog.

I definitely see the appeal of separating this now. Seems like a reasonable approach. We just need to make sure that we get the boostrap procedure right. There IMO shouldn't be any manual steps but it should be possible to boostrap such a "team" project through the config hierarchy, even if we wouldn't completely manage it after creation.